extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -73,7 +90,31 @@ function BaseCarouselWithRef(
   const scrollRef = useRef(null);
   const containRef = useRef(null);
   const contentRef = useRef(null);
+  const shouldAutoAdvance = useRef(autoAdvance);
+  const autoAdvanceTimesRef = useRef(0);
+  const autoAdvanceInterval = useMemo(
+    () => Math.max(customInterval, MIN_AUTO_ADVANCE_INTERVAL),
+    [customInterval]
+  );
 
+  const advance = useCallback(
+    (by, currentSlide) => {
+      if (
+        autoAdvanceTimesRef.current >=
```

Comment:
Could you please comment in code here on how the `autoAdvance` is stopped? Not quite sure the rules are very easy to pick up on from the code in this case.